src/app/component/general/eco-news/create-news/create-news.component.ts

```diff
@@ -1,18 +1,111 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, DoCheck } from '@angular/core';
 import { preparedImageForCreateEcoNews } from '../../../../links';
+import { Router } from '@angular/router';
+import { FormBuilder, Validators } from '@angular/forms';
+import { CreateEcoNewsService } from '../../../../service/eco-news/create-eco-news.service';
 
 @Component({
   selector: 'app-create-news',
   templateUrl: './create-news.component.html',
   styleUrls: ['./create-news.component.css']
 })
-export class CreateNewsComponent implements OnInit {
+export class CreateNewsComponent implements OnInit, DoCheck {
 
-  link: string = preparedImageForCreateEcoNews;
+  constructor(private router: Router, 
+              private fb: FormBuilder, 
+              private createEcoNewsService: CreateEcoNewsService) {}
 
-  constructor() { }
+  public createNewsForm = this.fb.group({
+    title: ['', [Validators.maxLength(170)]],
+    source: [''],
+    content: ['', [Validators.minLength(20)]],
+    tags: this.fb.array([])
+  });
+
+  public filters: Array<object> = [
```

Comment:
Please specify the type in generic with more details. For example 'Array<{name: string, isActive: boolean}>'.

Reply:
fixed